tests/topotests/bgp_vpnv4_per_nexthop_label/test_bgp_vpnv4_per_nexthop_label.py

```diff
@@ -0,0 +1,818 @@
+
+    switch = tgen.add_switch("s5")
+    switch.add_link(tgen.gears["r1"])
+    switch.add_link(tgen.gears["r14"])
+
+
+def _populate_iface():
+    tgen = get_topogen()
+    cmds_list = [
+        "ip link add vrf1 type vrf table 10",
+        "echo 100000 > /proc/sys/net/mpls/platform_labels",
+        "ip link set dev vrf1 up",
+        "ip link set dev {0}-eth1 master vrf1",
+        "echo 1 > /proc/sys/net/mpls/conf/{0}-eth0/input",
+    ]
+    cmds_list_plus = [
+        "ip link set dev {0}-eth2 master vrf1",
+    ]
+
+    for cmd in cmds_list:
+        input = cmd.format("r1")
+        logger.info("input: " + cmd)
+        output = tgen.net["r1"].cmd(cmd.format("r1"))
+        logger.info("output: " + output)
+
+    for cmd in cmds_list_plus:
+        input = cmd.format("r1")
+        logger.info("input: " + cmd)
+        output = tgen.net["r1"].cmd(cmd.format("r1"))
+        logger.info("output: " + output)
+
+    for cmd in cmds_list:
+        input = cmd.format("r2")
+        logger.info("input: " + cmd)
+        output = tgen.net["r2"].cmd(cmd.format("r2"))
+        logger.info("output: " + output)
+
+
+def setup_module(mod):
+    "Sets up the pytest environment"
+    tgen = Topogen(build_topo, mod.__name__)
+    tgen.start_topology()
+
+    router_list = tgen.routers()
+    _populate_iface()
+
+    for rname, router in router_list.items():
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_BGP, os.path.join(CWD, "{}/bgpd.conf".format(rname))
+        )
+
+    # Initialize all routers.
+    tgen.start_router()
+
+
+def teardown_module(_mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+
+    tgen.stop_topology()
+
+
+def bgp_vpnv4_table_check(router, group, labelList=None, label_value_expected=None):
+    """
+    Dump and check that vpnv4 entries have the same MPLS label value
+    * 'router': the router to check
+    * 'group': the list of prefixes to check. a single label value for the group has to be found
+    * 'labelList': check that the label values are not present in the vpnv4 entries
+    *              that list is updated with the present label value
+    * 'label_value_expected': check that the mpls label read is the same as that value
+    """
+
+    storedLabelInited = False
+    for entry in group:
+        dump = router.vtysh_cmd("show bgp ipv4 vpn {} json".format(entry), isjson=True)
+        for rd in dump:
+            assert (
+                "prefix" in dump[rd].keys()
+            ), "{0}, {1}, route distinguisher not present".format(router.name, entry)
+            assert dump[rd]["prefix"] == str(
+                entry
+            ), "{0}, {1}, ipv4 vpn entry not present".format(router.name, entry)
+            assert (
+                "paths" in dump[rd].keys()
+            ), "{0}, {1}, paths keyword not present}".format(router.name, entry)
```

Comment:
those three assert are not needed.
Furthermore, if the prefix is not found the answer is {}, which means you are not entering in the loop and nothing will fail.
```
vtysh -c 'show bgp ipv4 vpn 10.200.1.0/24 json' | jq .
{}
```
You need that outside the if

`assert dump, "{0}, {1}, route distinguisher not present".format(router.name, entry)`

```
>>> assert({})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> assert({"test":None})
```

Reply:
done